xbmc/FileItem.cpp

```diff
@@ -3533,17 +3534,42 @@ int CFileItem::GetVideoContentType() const
   return type;
 }
 
+CFileItem CFileItem::GetItemToPlay() const
+{
+  if (HasEPGInfoTag())
+  {
+    const CPVRChannelPtr channel(GetEPGInfoTag()->ChannelTag());
+    if (channel)
+      return CFileItem(channel);
+  }
+  return CFileItem(*this);
```

Comment:
Even better would be to potentially use `CFileItemPtr` instead, but that probably requires more careful thought.